packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -185,6 +185,26 @@ class _DefaultPub implements Pub {
       _fileSystem.path.join(directory, 'pubspec.lock')
     );
 
+    if (packageConfigFile.existsSync()) {
```

Comment:
I don't think this is the right place to do it. Consider the case when someone has used some other tool to generate a package config but wants to switch back to pub. They will be unable to do it until they run flutter clean, even if they manually run `flutter pub get`.

Instead I think you should either add this logic to the flutter_command code that determines whether to run pub or add a new parameter that distinguishes between an automatic pub get and a pub get that was explicitly asked for

Reply:
Fair enough. Where would be a correct place to do it exactly then? Because there are multiple commands doing the same thing

Reply:
In that case I would probably go the route of distinguishing between automatic and manual pub gets via a parameter to this function, and then only skipping the pub get if it was automatic

Reply:
I have pushed a similar flag.